pipelines/scipipe/obs_lsst.groovy

```diff
@@ -0,0 +1,405 @@
+    // butler
+    def runSlug = "${datasetSlug(conf)}^${codeSlug(conf)}"
+
+    matrix[runSlug] = {
+      obsLsstDataset(
+        config: conf,
+        dockerImage: dockerImage,
+        slug: runSlug,
+        wipeout: wipeout,
+      )
+    }
+  }
+
+  stage("${jobConfName} matrix") {
+    parallel(matrix)
+  }
+} // notify.wrap
+
+/**
+ * Generate a "slug" to describe this dataset including branch name.
+ *
+ * @param conf Map
+ */
+def String datasetSlug(Map conf) {
+  def ds = conf.dataset
+  def slug = "${ds.display_name}-${ds.git_ref.tr('/', '_')}"
+  return slug.toLowerCase()
+}
+
+/**
+ * Generate a "slug" to describe the obs_lsst code including branch name.
+ *
+ * @param c Map
+ */
+def String codeSlug(Map conf) {
+  def code = conf.code
+
+  def name = 'obs_lsst'
+  def ref = 'installed'
+
+  if (code) {
+    name = code.display_name
+  }
+  if (code?.github_repo) {
+    name = code.name
+    ref  = code.git_ref.tr('/', '_')
+  }
+  def slug = "${name}-${ref}".toLowerCase()
+  return slug.toLowerCase()
+}
+
+/**
+ * `name` to use to refer to dataset/code
+ *
+ * @param m Map
+ */
+def String displayName(Map m) {
+  m.display_name ?: m.name
+}
+
+/**
+ * Prepare, execute, and record results of a obs_lsst CI run.
+ *
+ * @param p Map
+ * @param p.config Map Dataset configuration.
+ * @param p.dockerImage String
+ * @param p.slug String Name of dataset.
+ * @param p.wipeout Boolean
+ */
+def void obsLsstDataset(Map p) {
+  util.requireMapKeys(p, [
+    'config',
+    'dockerImage',
+    'slug',
+    'wipeout',
+  ])
+
+  def conf = p.config
+  def ds   = conf.dataset
+  def code = conf.code
+
+  // code.name is required in order to build code
+  Boolean buildCode = code?.name
+
+  def run = {
+    // note that pwd() must be run inside of a node {} block
+    def jobDir           = pwd()
+    def datasetDir       = "${jobDir}/datasets/${ds.name}"
+    def ciDir            = "${jobDir}/ci-scripts"
+    def baseDir          = "${jobDir}/${p.slug}"
+    // the code clone needs to be under the long winded path for archiving
+    def codeDir          = buildCode ? "${baseDir}/${code.name}" : ''
+    def homeDir          = "${baseDir}/home"
+    def runDir           = "${baseDir}/run"
+    def fakeLsstswDir    = "${baseDir}/lsstsw-fake"
+
+    docker.image(p.dockerImage).pull()
+    def labels = util.shJson """
+      docker inspect --format '{{json .Config.Labels }}' ${p.dockerImage}
+    """
+
+    if (!labels.VERSIONDB_MANIFEST_ID) {
+      missingDockerLabel 'VERSIONDB_MANIFEST_ID'
+    }
+    if (!labels.LSST_COMPILER) {
+      missingDockerLabel 'LSST_COMPILER'
+    }
+
+    String manifestId = labels.VERSIONDB_MANIFEST_ID
+    String lsstCompiler = labels.LSST_COMPILER
+
+    // empty ephemeral dirs at start of build
+    util.emptyDirs([
+      homeDir,
+      runDir,
+    ])
+
+    // stage manifest.txt early so we don't risk a long processing run and
+    // then fail setting up to run dispatch_verify.py
+    stageFakeLsstsw(
```

Comment:
The "fake" `manifest.txt/repos.yaml` are only needed if `dispatch_verify` is going to be run.  Are you intending to eventually push results to squash?